### PR TITLE
Standardise arguments in convert/generate scripts

### PR DESCRIPTION
**Pull request by SamPUG**
This pull request contains changes related to issue #18 .

I have standardised the input arguments for downloading and extracting cross-section data. This involved removing the --batch flag and adding the `--no-download` `--download` `--no-extract` and `--extract` flags in the following files:
- convert_fendl.py
- convert_tendl.py 
- generate_cendl.py
- generate_gendl.py 

The convert_jeff32.py script was used as reference script for the behavior. 

The following tests were carried out to ensure the changes function as expected:
- Run each script with the new flags to ensure that they produce the correct behavior
-  For the convert scripts, check the final hdf5 files against those generated by the old version using h5diff
- For generate scripts, the conversion part of the run was skipped due to the amount of time it takes. Instead the endf files were compared. Note that there were no changes made to the conversion process so no difference should be expected.

### CODE REVIEW

**Comment by Shimwell** [comment]
Thanks for this PR, I can take a look at these and run them when I get back to the office on Monday

File: convert_tendl.py [hunk 1]
```diff
@@ -76,36 +84,26 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
-        continue
-
-    # Extract files
-    with tarfile.open(f, 'r') as tgz:
-        print('Extracting {}...'.format(f))
-        tgz.extractall(path=ace_files_dir)
+if args.extract:
+    for f in release_details[args.release]['files']:
+        with tarfile.open(f, 'r') as tgz:
+            print('Extracting {0}...'.format(f))
```
Comment (paulromano) [inline convert_tendl.py]:
```suggestion
            print(f'Extracting {f}...'.)
```

File: convert_fendl.py [hunk 2]
```diff
@@ -67,38 +76,27 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
-    # Establish connection to URL
-    url = release_details[args.release]['base_url'] + f
-    downloaded_file = download(url, as_browser=True,
-                               context=ssl._create_unverified_context())
-    files_complete.append(downloaded_file)
+if args.download:
+    print(download_warning)
+    for f in release_details[args.release]['files']:
+        download(urljoin(release_details[args.release]['base_url'], f),
+                    as_browser=True, context=ssl._create_unverified_context())
```
Comment (paulromano) [inline convert_fendl.py]:
```suggestion
                 as_browser=True, context=ssl._create_unverified_context())
```

File: convert_fendl.py [hunk 3]
```diff
@@ -67,38 +76,27 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
 
 # ==============================================================================
 # EXTRACT FILES FROM TGZ
+if args.extract:
```
Comment (paulromano) [inline convert_fendl.py]: Looks like the comment above needs to be fixed: TGZ -> ZIP

**Comment by SamPUG** [comment]
Thanks for the suggested changes @paulromano. I've made them all.

**Comment by Shimwell** [comment]
I have now used the convert_fendl.py and convert_tendl.py scripts and they both work as described

**Pull request merged**